specs/NNNN-template/README.md

```diff
@@ -0,0 +1,161 @@
+
+# VEP-NNNN: Your short, descriptive title
+
+* **Author(s):** Name (mail), ...
+* **Status:** draft | implemented | rejected | withdrawn | replaced
+
+To get started with this template:
+
+- [ ] **Create an issue in Github (if one does not exists already)**
+- [ ] **Make a copy of this template directory.**
+  Copy this template into the specs directory and name it
+  `NNNN-short-descriptive-title`, where `NNNN` is the issue number (with no
+  leading-zero padding) created above.
+- [ ] **Fill out this file as best you can.**
+  There are instructions as HTML comments.
+  At minimum, you should fill in the "Summary" and "Motivation" sections.
+- [ ] **Create a PR for this VEP.**
+- [ ] **Merge early and iterate.**
+  Avoid getting hung up on specific details and instead aim to get the goals of
+  the VEP clarified and merged quickly. The best way to do this is to just
+  start with the high-level sections and fill out details incrementally in
+  subsequent PRs.
+
+<!-- Provide table of content as it's helpful. -->
+
+- [Summary](#summary)
+- [Glossary](#glossary)
+- [Motivation](#motivation)
+- [Requirements and Goals](#requirements-and-goals)
+- [System Interface (API)](#system-interface--api-)
+- [High-level design](#high-level-design)
+- [Detailed design](#detailed-design)
+- [Security and Permissions](#security-and-permissions)
+- [Implementation Stories](#implementation-stories)
+- [Alternatives](#alternatives)
+
+## Summary
+
+<!--
+Short summary of the proposal. It will be used as user-focused
+documentation such as release notes or a (customer facing) development roadmap.
+The tone and content of the `Summary` section should be
+useful for a wide audience.
+-->
+
+## Glossary
+<!--
+Optional section which defines terms and abbreviations used in the rest of the document.
+-->
+
+## Motivation
```

Comment:
I think we can merge _Motivation_ and _Requirements and Goals_ into one section (e.g. _Problem definition and goals_)

Reply:
I decided to keep them separate because they answer different questions. 
Motivation answers the question why do we need this? 
And requirements & goals answer the question what do we need ?
...

Kubernetes and Python specs do merge them in one section. So there's an argument for that.

Reply:
Motivation explains why are doing this . 
Depending on what you mean by use-cases if they explain the user problem they'd be in the motivation section, 
If they explain what we do to solve it they'd be in requirements. 


I have started creating an example here - https://github.com/vmware/versatile-data-kit/pull/730

Also this is shaping to be good example also - https://github.com/vmware/versatile-data-kit/pull/748

Since a lot of this spec is based on KEPs, there are some very good examples in Kubernetes KEPS: Those are some fairly simple and understandable: 

https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1898-hardened-exec#motivation or https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1539-hugepages#motivation